console/src/app/components/features/features.component.ts

```diff
@@ -78,140 +98,59 @@ export class FeaturesComponent {
     ];
     this.breadcrumbService.setBreadcrumb(breadcrumbs);
 
-    this.getFeatures(true);
+    this.getFeatures();
   }
 
   public validateAndSave() {
-    this.featureService.resetInstanceFeatures().then(() => {
-      const req = new SetInstanceFeaturesRequest();
-      let changed = false;
-
-      console.log(this.toggleStates);
-
-      if (this.toggleStates?.loginDefaultOrg?.state !== ToggleState.INHERITED) {
-        req.setLoginDefaultOrg(this.toggleStates?.loginDefaultOrg?.state === ToggleState.ENABLED);
-        changed = true;
-      }
-      if (this.toggleStates?.oidcTriggerIntrospectionProjections?.state !== ToggleState.INHERITED) {
-        req.setOidcTriggerIntrospectionProjections(
-          this.toggleStates?.oidcTriggerIntrospectionProjections?.state === ToggleState.ENABLED,
-        );
-        changed = true;
-      }
-      if (this.toggleStates?.oidcLegacyIntrospection?.state !== ToggleState.INHERITED) {
-        req.setOidcLegacyIntrospection(this.toggleStates?.oidcLegacyIntrospection?.state === ToggleState.ENABLED);
-        changed = true;
-      }
-      if (this.toggleStates?.userSchema?.state !== ToggleState.INHERITED) {
-        req.setUserSchema(this.toggleStates?.userSchema?.state === ToggleState.ENABLED);
-        changed = true;
-      }
-      if (this.toggleStates?.oidcTokenExchange?.state !== ToggleState.INHERITED) {
-        req.setOidcTokenExchange(this.toggleStates?.oidcTokenExchange?.state === ToggleState.ENABLED);
-        changed = true;
-      }
-      if (this.toggleStates?.actions?.state !== ToggleState.INHERITED) {
-        req.setActions(this.toggleStates?.actions?.state === ToggleState.ENABLED);
-        changed = true;
-      }
-      if (this.toggleStates?.oidcSingleV1SessionTermination?.state !== ToggleState.INHERITED) {
-        req.setOidcSingleV1SessionTermination(
-          this.toggleStates?.oidcSingleV1SessionTermination?.state === ToggleState.ENABLED,
-        );
-        changed = true;
-      }
-
-      if (changed) {
-        this.featureService
-          .setInstanceFeatures(req)
-          .then(() => {
-            this.toast.showInfo('POLICY.TOAST.SET', true);
-          })
-          .catch((error) => {
-            this.toast.showError(error);
-          });
-      }
+    const req = create(SetInstanceFeaturesRequestSchema, {
```

Comment:
the create is not needed if you want type hints for example you can just use const req: MessageInitShape<typeof SetInstanceFeaturesRequestSchema> = {...}

Reply:
when removing the create I get a `Property '$typeName' is missing in type...`

Reply:
answered why this is the case in my other comment